actions/slack-notify-deploy/action.yml

```diff
@@ -0,0 +1,118 @@
+name: Slack Notify Deployment
+description: Send slack notifications about deployments
+
+inputs:
+  slack_token:
+    required: true
+    description: Token for github requests.
+  env:
+    required: true
+    description: Environment to deploy to
+  commit_data:
+    required: false
+    description: Data from commits-range-action.
+  started:
+    required: false
+    description: Report started instead of job status.
+  unstable:
+    required: false
+    description: Report started instead of job status.
+
+runs:
+  using: "composite"
+  steps:
+    - name: Prepare Slack message
+      uses: actions/github-script@v6
+      id: prepare_payload
+      env:
+        COMMIT_DATA: ${{ inputs.commit_data }}
+        ENV_NAME: ${{ inputs.env }}
+        JOB_STATUS: ${{ job.status }}
+        STARTED: ${{ inputs.started }}
+        UNSTABLE: ${{ inputs.unstable }}
+      with:
+        script: |
+          const data = JSON.parse(process.env.COMMIT_DATA)
+          const jobStatus = process.env.STARTED == 'true' ? 'started' : process.env.JOB_STATUS;
```

Comment:
Where do you get the `JSON` or the `process` from?

Reply:
This action allows you to write node.js code to init the workflow (both `process` and `JSON` are globals there)

Reply:
Shouldn't `JOB_STATUS` include the `STARTED` value?

Reply:
nope, it equals `success` until is equals something else

the reasoning for that to my best understanding is that every step has a default `if` value that is `${{ success() }}`
if you want a step to always run, use `if: ${{ always() }}`